packages/nftoupon/src/Creator.tsx

```diff
@@ -22,597 +24,664 @@ import confetti from "canvas-confetti";
+
+      if (!isEmpty(payload)) {
+        setXummPayload(payload);
+        setVisible(true);
+      } else {
+        setXummPayload(null);
+      }
+    } catch (error) {
+      toast.error(ERROR_IN_API);
+    }
+
+    setIsLoading(false);
+  };
+
+  const closeSocket = (ws: WebSocket) => {
+    ws.close();
+    setXummPayload(null);
+    setVisible(false);
+  };
+
+  useEffect(() => {
+    const getDetails = async () => {
+      try {
+        const options = { address: walletAddress };
+        const { nftoupons } = await fetcher(
+          NFToupon_Key,
+          GET_META_URL,
+          options
+        );
+        setData(nftoupons);
+      } catch (error) {
+        toast.error(ERROR_IN_API);
+      }
+    };
+    if (!isEmpty(walletAddress)) {
+      getDetails();
+    }
+  }, [walletAddress, transactionType, NFToupon_Key]);
+
+  const signValidator = async (option: any, ws: WebSocket) => {
+    const result = await fetcher(NFToupon_Key, CARGO_URL, option);
+    setWalletAddress(result?.payload);
+    setTransactionType(result?.tx_type);
+    closeSocket(ws);
+  };
+
+  useEffect(() => {
+    if (!isEmpty(xummPayload)) {
+      const wsURL = xummPayload?.refs?.websocket_status;
+      const ws = new WebSocket(wsURL || "");
+      ws.onmessage = (event) => {
+        const { opened, payload_uuidv4, signed, expired } = JSON.parse(
+          event.data
+        );
+        if (opened) {
+        } else if (expired) {
+          closeSocket(ws);
+          toast.error(EXPIRED);
+        } else if (!isEmpty(payload_uuidv4) && !signed) {
+          closeSocket(ws);
+          toast.error(FAIL_SIGN);
+        } else if (signed) {
+          const option = { payload_uuidv4 };
+          signValidator(option, ws);
+        }
+      };
+    }
+  }, [xummPayload, NFToupon_Key]);
+
+  useEffect(() => {
+    if (transactionType === "NFTokenMint") {
+      try {
+        const saveTokens = async () => {
+          const options = {
+            address: walletAddress,
+            title: "",
+            description: textAreaValue,
+            status: "Pending",
+            imageUrl: xummPayload?.imageUrl,
+          };
+          await fetcher(NFToupon_Key, NFTOKEN_MINT_URL, options);
+        };
+        saveTokens();
+        handleConfetti();
+      } catch (error) {
+        toast.error(ERROR_IN_API);
+      }
+    } else if (transactionType === "NFTokenAcceptOffer") {
+      try {
+        const updateTokenStatus = async () => {
+          const options = { status: "Accepted", id: details.id };
+          await fetcher(NFToupon_Key, NFTOKEN_ACCEPT_OFFER_URL, options);
+        };
+        updateTokenStatus();
+      } catch (error) {
+        toast.error(ERROR_IN_API);
+      }
+    }
+    if (
+      transactionType === "NFTokenAcceptOffer" ||
+      transactionType === "NFTokenMint"
+    ) {
+      // reset to default
+      setTextAreaValue("");
+      setDetails(DETAILS);
```

Comment:
are you sure this will work... if I click on one of the image it won't reset the default value which is incorrect?

Reply:
yes working for me